### PR TITLE
Add a CLI arg to CleanUpM2Repository to accept an arbitrary path for the .m2 folder

### PR DESCRIPTION
**Pull request by mir-am**
This is a very small PR to clean up the .m2 folder at arbitrary paths.
Note that in production we store POM files at a different path not in the home folder as K8s pods store data at `/mnt/fasten/data`.

### CODE REVIEW

File: plugins/pom-analyzer/src/main/java/eu/f4sten/pomanalyzer/CleanUpM2Repository.java [hunk 1]
```diff
@@ -32,15 +33,19 @@
...
-    private static final String BASE_DIR = "~/.m2/repository";
+    @Inject
+    public CleanUpM2Repository(CleanUpM2RepositoryArgs args) {
+        this.args = args;
+    }
 
     @Override
     public void run() {
-        LOG.info("Searching for invalid pom.xml files in: {}", BASE_DIR);
+        LOG.info("Searching for invalid pom.xml files in: {}", this.args.m2RepositoryPath);
```
Comment (proksch) [inline plugins/pom-analyzer/src/main/java/eu/f4sten/pomanalyzer/CleanUpM2Repository.java:45]: Please avoid unnecessary `this`

File: plugins/pom-analyzer/src/main/java/eu/f4sten/pomanalyzer/CleanUpM2Repository.java [hunk 2]
```diff
@@ -32,15 +33,19 @@
...
 
         try {
-            Files.walkFileTree(Path.of(BASE_DIR), new SimpleFileVisitor<Path>() {
+            Files.walkFileTree(Path.of(this.args.m2RepositoryPath), new SimpleFileVisitor<Path>() {
```
Comment (proksch) [inline plugins/pom-analyzer/src/main/java/eu/f4sten/pomanalyzer/CleanUpM2Repository.java:48]: Please avoid unnecessary `this`

File: plugins/pom-analyzer/src/main/java/eu/f4sten/pomanalyzer/CleanUpM2RepositoryArgs.java [hunk 3]
```diff
@@ -0,0 +1,10 @@
+package eu.f4sten.pomanalyzer;
+
+import com.beust.jcommander.Parameter;
+
+public class CleanUpM2RepositoryArgs {
+
+    @Parameter(names= "--m2repo.path", arity = 1)
```
Comment (proksch) [inline plugins/pom-analyzer/src/main/java/eu/f4sten/pomanalyzer/CleanUpM2RepositoryArgs.java:7]: Please rename parameter to `--path.m2`

File: plugins/pom-analyzer/src/main/java/eu/f4sten/pomanalyzer/CleanUpM2RepositoryArgs.java [hunk 4]
```diff
@@ -0,0 +1,10 @@
...
+    public String m2RepositoryPath = "~/.m2/repository";
```
Comment (proksch) [inline plugins/pom-analyzer/src/main/java/eu/f4sten/pomanalyzer/CleanUpM2RepositoryArgs.java:8]: Please rename field to `pathM2` (to stay consistent with the naming of the argument.)

**Review by proksch — CHANGES_REQUESTED** [review]
Generally looks good to me, but I have minor requests for renaming.

**Comment by mir-am** [comment]
Thanks for the comments.
I have addressed them in https://github.com/fasten-project/data-processing/pull/17/commits/d44dd0d4ce03c84c451e545090e4c33740dc3318.

**Review by proksch — APPROVED** [review]

**Pull request merged**